### PR TITLE
migrate_options_shared: Add pty channel migration

### PR DESCRIPTION
**Pull request by dzhengfy**
Test migration with pty channel is working well.

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/2026

**Comment by dzhengfy** [comment]
(1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.pty_channel.without_postcopy: PASS (148.26 s)

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 1]
```diff
@@ -347,6 +314,27 @@ def do_actions_during_migrate(params):
                 check_domjobinfo(params)
             time.sleep(3)
 
+    def add_channel_xml():
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_options_shared.py]: add_channel_xml change to attach_channel_xml ?

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 2]
```diff
@@ -624,6 +641,46 @@ def check_migration_res(result):
 
         check_migration_res(mig_result)
 
+        if add_channel:
+            # Get the channel device source path of remote guest
+            if not remote_virsh_session:
+                remote_virsh_session = virsh.VirshPersistent(**remote_virsh_dargs)
+            file_path = tempfile.mktemp(dir=data_dir.get_tmp_dir())
```
Comment (kylazhang) [inline libvirt/tests/src/migration/migrate_options_shared.py]: should be file_path = data_dir.get_tmp_dir()

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import re
 import threading
 import platform
+import tempfile
```
Comment (kylazhang) [inline libvirt/tests/src/migration/migrate_options_shared.py]: should be useless

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 4]
```diff
@@ -506,12 +508,14 @@ def check_migration_res(result):
...
-            new_xml.remove_all_device_by_type('controller')
-            logging.debug("After removing controllers, current XML:\n%s\n", new_xml)
-            add_ctrls(new_xml, dev_index=contrl_index)
+        # Create a remote runner for later use
+        runner_on_target = remote.RemoteRunner(host=server_ip,
+                                               username=server_user,
+                                               password=server_pwd)
 
+        # Change the configuration files if needed before starting guest
+
```
Comment (kylazhang) [inline libvirt/tests/src/migration/migrate_options_shared.py]: Remove the blank line

Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Updated

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: This is to create a temp file in the temp directory.

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 3 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: I use it to create a temp file.

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 2 above)
Comment (kylazhang) [reply libvirt/tests/src/migration/migrate_options_shared.py]: ok, thanks

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: updated

**Pull request merged**